oxenss/rpc/request_handler.cpp

```diff
@@ -1049,6 +1049,69 @@ void RequestHandler::process_client_req(
         reply_or_fail(std::move(res));
 }
 
+void RequestHandler::process_client_req(
+        rpc::revoked_subaccounts&& req, std::function<void(Response)> cb) {
+    log::debug(logcat, "processing revoked_subaccounts request");
+
+    if (!service_node_.is_pubkey_for_us(req.pubkey))
+        return cb(handle_wrong_swarm(req.pubkey));
+
+    auto now = system_clock::now();
+    if (req.timestamp < now - SIGNATURE_TOLERANCE || req.timestamp > now + SIGNATURE_TOLERANCE) {
+        log::debug(
+                logcat,
+                "revoked_subaccounts: invalid timestamp ({}s from now)",
+                duration_cast<seconds>(req.timestamp - now).count());
+        return cb(Response{
+                http::NOT_ACCEPTABLE, "revoked_subaccounts timestamp too far from current time"sv});
+    }
+
+    if (!verify_signature(
+                service_node_.get_db(),
+                req.pubkey,
+                std::nullopt,
+                std::nullopt,  // no subaccount allowed
+                subaccount_access::None,
+                false,
+                req.signature,
+                "revoked_subaccounts",
+                req.timestamp)) {
+        log::debug(logcat, "revoked_subaccounts: signature verification failed");
+        return cb(Response{
+                http::UNAUTHORIZED, "revoked_subaccounts signature verification failed"sv});
+    }
+
+    std::vector<std::string> revoked_subaccounts;
+    try {
+        revoked_subaccounts = service_node_.get_db().revoked_subaccounts(req.pubkey);
+    } catch (const std::exception& e) {
+        auto msg = fmt::format(
+                "Internal Server Error. Could not retrieve revoked_subaccounts for {}",
+                obfuscate_pubkey(req.pubkey));
+        log::critical(logcat, "{}", msg);
+        return cb(Response{http::INTERNAL_SERVER_ERROR, std::move(msg)});
+    }
+
+    log::trace(
+            logcat,
+            "Retrieved {} revoked_subaccounts for {}",
+            revoked_subaccounts.size(),
+            obfuscate_pubkey(req.pubkey));
+
+    json messages = json::array();
+    for (const auto& revoked_subaccount : revoked_subaccounts) {
+        messages.push_back(json{
+                {"token_hex", oxenc::to_hex(revoked_subaccount)},
+                {"token_b64", oxenc::to_base64(revoked_subaccount)},
+        });
```

Comment:
I would return this as only b64 rather than the two different encodings of the same value.  Also if `req.b64` is not set then it's a request that wants raw bytes rather than encoded values (generally that means a bt-encoded request -- in such a case, we are only using the json value here a container: when we do the final response processing for those requests we translate the stuff inside the json object into a bt-encoded return value, rather than dumping the json):
```suggestion
        messages.push_back(json{
                {"token", req.b64 ? oxenc::to_base64(revoked_subaccount) : revoked_subaccount},
        });
```

I think that you could also go a step further, by just returning an array of values (instead of an array of objects):

```suggestion
        messages.push_back(req.b64 ? oxenc::to_base64(revoked_subaccount) : revoked_subaccount);
```
but I don't feel strongly about that, and it's a bit more future-proof to not do that if we conceivable want to add other values here in the future.

Reply:
isn't this going to crash the json stringifier?
`revoked_subaccount` is a blob, which might not be a valid string, and that was an issue I had.

should it maybe be
`messages.push_back(req.b64 ? oxenc::to_base64(revoked_subaccount) : to_hex(revoked_subaccount));` instead?

Reply:
> isn't this going to crash the json stringifier?

No; when `.b64` is not false, we don't stringify at all.